common/src/main/java/feast/common/function/StringUtils.java

```diff
@@ -0,0 +1,211 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.common.function;
+
+import feast.proto.core.FeatureSetProto.FeatureSetSpec;
+import feast.proto.core.FeatureSetReferenceProto.FeatureSetReference;
+import feast.proto.core.StoreProto;
+import feast.proto.serving.ServingAPIProto.FeatureReference;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class StringUtils {
+
+  private static String PROJECT_DEFAULT_NAME = "default";
+
+  // FeatureSet-related functionality
+  /**
+   * Accepts FeatureSetSpec object and returns its reference in String "project/featureset_name".
+   *
+   * @param featureSetSpec
+   * @return String format of FeatureSetReference
+   */
+  public static String getFeatureSetStringRef(FeatureSetSpec featureSetSpec) {
+    return String.format("%s/%s", featureSetSpec.getProject(), featureSetSpec.getName());
+  }
+
+  /**
+   * Accepts FeatureSetReference object and returns its reference in String
+   * "project/featureset_name".
+   *
+   * @param featureSetReference
+   * @return String format of FeatureSetReference
+   */
+  public static String getFeatureSetStringRef(FeatureSetReference featureSetReference) {
+    return String.format("%s/%s", featureSetReference.getProject(), featureSetReference.getName());
+  }
+
+  // Feature-related functionality
+  /**
+   * Accepts FeatureReference object and returns its reference in String
+   * "project/featureset_name:feature_name".
+   *
+   * @param featureReference
+   * @return String format of FeatureReference
+   */
+  public static String getFeatureStringRef(
+      FeatureReference featureReference, boolean ignoreProject) {
+    String ref = featureReference.getName();
+    if (!featureReference.getFeatureSet().isEmpty()) {
+      ref = featureReference.getFeatureSet() + ":" + ref;
+    }
+    if (!featureReference.getProject().isEmpty() && !ignoreProject) {
+      ref = featureReference.getProject() + "/" + ref;
+    }
+    return ref;
+  }
+
+  // Subscription-related functionality
+  /**
+   * Accepts a comma-delimited Subscriptions that is string-formatted and converts it to a list of
+   * Subscription class objects.
+   *
+   * @param subscriptions String formatted Subscriptions, comma delimited.
+   * @param exclude flag to determine if subscriptions with exclusion flag should be returned
+   * @return List of Subscription class objects
+   */
+  public static List<StoreProto.Store.Subscription> getSubscriptionsByStr(
```

Comment:
maybe better name "parseSubscriptionFrom"? similarly to protobuf api https://developers.google.com/protocol-buffers/docs/reference/java/com/google/protobuf/Parser

Reply:
Renamed function to follow protobuf api convention.